partiql-parser/src/result.rs

```diff
@@ -4,33 +4,140 @@
 
 use pest::error::{ErrorVariant, LineColLocation};
 use std::fmt;
+use std::fmt::Formatter;
 use thiserror::Error;
 
-/// Position in the source for an error.
+/// A line and column location.
+///
+/// This value is one-based, as that is how most people think of lines and columns.
+///
+/// ## Example
+/// ```
+/// # use partiql_parser::prelude::*;
+/// println!("Beginning of a document: {}", LineAndColumn::at(1, 1));
+/// ```
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub struct LineAndColumn(pub usize, pub usize);
+
+impl LineAndColumn {
+    /// Constructs a [`LineAndColumn`].
+    #[inline]
+    pub fn at(line: usize, column: usize) -> Self {
+        Self(line, column)
+    }
+
+    /// Returns a [`LineAndColumn`] that moves to the given position relative
+    /// to this one as a sort of "origin."
+    ///
+    /// Note that this positioning is 1-based, so moving `(1, 1)` to `(1, 1)` is a no-op.
```

Comment:
Why is the move 1-based? "Move 1 line" not moving 1 line seems weird

Reply:
Maybe the name is not right--this isn't taking a delta, change `location` to use `self` as the origin.  I am not sure what I should call that, but `move_relative` is not right--let me think about that.

Reply:
ah, ok